### PR TITLE
feat: added Number param

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: src/param.js [hunk 1]
```diff
@@ -336,3 +336,23 @@ class Json extends Param {
   }
 }
 module.exports.Json = (...args) => new Json(...args);
+
+class NumberParam extends Json {
+  constructor(name, { positive = false } = {}, ...args) {
```
Comment (MacMcIrish) [inline src/param.js]:
```suggestion
  constructor(name, { enforcePositive = false } = {}, ...args) {
```
?

Comment (MacMcIrish) [reply src/param.js]: Things like `min, max` etc can come later

File: test/param.spec.js [hunk 2]
```diff
@@ -290,6 +290,56 @@ describe("Testing Params", () => {
...
+        param: '-12.34'
+      }
+    })).to.throw('Invalid Value for query-Parameter "param" provided.');
+  });
+
+  it("Testing Number Parameter (json)", () => {
+    const param = api.Number("param", {}, "json");
+    expect(param.get({
+      body: {
+        param: -12.34
```
Comment (MacMcIrish) [inline test/param.spec.js]: There is an issue here, `'-12.34'` (note the string) is also a valid number apparently?

File: src/param.js (hunk 1 above)
Comment (simlu) [reply src/param.js]: `nonNegative: false`

Comment (simlu) [reply src/param.js]: Let's just replace this with min max

**Pull request merged**